src/exchange/disclaimer/disclaimer.controller.js

```diff
@@ -28,44 +25,73 @@ angular
         }
         /* eslint-enable class-methods-use-this */
 
-        /* eslint-disable class-methods-use-this */
-        hasFullSlot (list) {
-            return _.some(list, (item) => !item.emptySlotFlag);
+        _getDisclaimers ({ pageSize, offset }) {
+            return this.services
+                .Exchange.getDisclaimers(this.organization, this.productId, pageSize, offset - 1);
         }
-        /* eslint-enable class-methods-use-this */
-
-        loadPaginated (count, offset) {
-            this.loading.table = true;
 
-            this.services
-                .Exchange.getDisclaimers(this.$routerParams.organization, this.$routerParams.productId, count, offset)
+        refreshList () {
+            if (!this.disclaimersList) { return undefined; }
+            const config = { pageSize: this.pageSize, offset: this.offset };
+            return this._getDisclaimers(config)
                 .then((disclaimers) => {
-                    this.disclaimersList = disclaimers;
-                    this.setMessagesFlags(disclaimers);
+                    for (let i = 0; i < disclaimers.list.results.length; i++) {
```

Comment:
Are you just copying the array ? Can't you use assignation, the spread syntax, `_.clone` or `_.cloneDeep` if you need to ?

Reply:
No i'm modifying it in place, which no Lodash function is able to do.